frb_codegen/src/parser/ty.rs

```diff
@@ -157,32 +174,39 @@ impl<'a> TypeParser<'a> {
         generic: SupportedInnerType,
         _len: usize,
     ) -> Option<IrType> {
-        self.convert_to_ir_type(generic).map(|inner| match inner {
-            Primitive(primitive) => PrimitiveList(IrTypePrimitiveList { primitive }),
-            others => GeneralList(IrTypeGeneralList {
-                inner: Box::new(others),
-            }),
-        })
+        self.convert_to_ir_type(generic, false)
+            .map(|inner| match inner {
+                Primitive(primitive) => PrimitiveList(IrTypePrimitiveList { primitive }),
+                others => GeneralList(IrTypeGeneralList {
+                    inner: Box::new(others),
+                }),
+            })
     }
 
     /// Converts a path type into an `IrType` if possible.
-    pub fn convert_path_to_ir_type(&mut self, p: SupportedPathType) -> Option<IrType> {
+    pub fn convert_path_to_ir_type(
+        &mut self,
+        mut p: SupportedPathType,
+        is_exception: bool,
+    ) -> Option<IrType> {
         let p_as_str = format!("{}", &p);
         let ident_string = &p.ident.to_string();
-        if let Some(generic) = p.generic {
+        if !p.generic.is_empty() {
             match ident_string.as_str() {
                 "SyncReturn" => {
                     // Special-case SyncReturn<Vec<u8>>. SyncReturn for any other type is not
                     // supported.
-                    match *generic {
+                    match &p.generic[0] {
```

Comment:
nit: Maybe we should ensure it has only *one* argument, e.g. `p.generic.single()`

Reply:
there's no `.single` on Vec

Reply:
We can add one in some utils.rs :)